motoman_driver/MotoPlus/IoServer.c

```diff
@@ -357,34 +365,40 @@ int Ros_IoServer_ReadIOGroup(SimpleMsg* receiveMsg, SimpleMsg* replyMsg)
 	replyMsg->header.msgType = ROS_MSG_MOTO_READ_IO_GROUP_REPLY;
 	replyMsg->header.commType = ROS_COMM_SERVICE_REPLY;
 
-	for (i = 0; i < 8; i += 1)
+	if (Ros_IoServer_IsValidReadAddress(receiveMsg->body.readIOGroup.ioAddress, IO_ACCESS_GROUP))
 	{
-		ioReadInfo[i].ulAddr = (receiveMsg->body.readIOGroup.ioAddress * 10) + i;
-	}
-	apiRet = mpReadIO(ioReadInfo, ioValue, 8);
+		for (i = 0; i < 8; i += 1)
+		{
+			ioReadInfo[i].ulAddr = (receiveMsg->body.readIOGroup.ioAddress * 10) + i;
```

Comment:
I may have asked this before, but why the `* 10`?

Reply:
When reading a whole 'group', it is expecting the group-number, which is the first four digits of an I/O address.
For example, for the signal 12345, it is Group 1234, Bit 5.